### ISSUE TITLE
Icon rendering problem in svg foreignObject since hass >= 0.110: icon "shifts" down vertically depending on zoom level & browser

### ISSUE THREAD

**Issue by AmoebeLabs** [home-assistant/frontend#7224]
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

As of hass 0.110 the rendering of mdc icons have changed. You have to add `--mdc-icon-size": '100%;'` to the css to get the icon rendered correctly within its parent.

However, I now have problems when zooming in: for some reason the vertical size of the `div` changes. As a result the icon shifts downward (is centered), and gets clipped by the parent.

Screenshot @300% zoom in Chrome of hass 0.98:
![image](https://user-images.githubusercontent.com/7996307/95014406-e1297500-0646-11eb-8e8c-7ea8d78ae41e.png)

Screenshot @300% zoom in Chrome of hass 0.115.6 (ignore the second set of icons, just for testing):
![image](https://user-images.githubusercontent.com/7996307/95014421-fdc5ad00-0646-11eb-96a8-b6a5657c667d.png)

On the mobile side: IOS 14 clips most of the icons when initially rendered.

This is the formatting:
![image](https://user-images.githubusercontent.com/7996307/95014485-80e70300-0647-11eb-9c90-989fd18180ce.png)

The foreignObject:
![image](https://user-images.githubusercontent.com/7996307/95014490-91977900-0647-11eb-877b-32c7a1396637.png)

And the weird height of the div:
![image](https://user-images.githubusercontent.com/7996307/95014498-a7a53980-0647-11eb-9d7f-11c5a623c110.png)

...causing the icon to be clipped.

I can't find WHY the height of the div becomes larger when zooming in!

## Expected behavior

Icons are rendered within the foreignObject.

Zoom @100%:
foreignObject:
![image](https://user-images.githubusercontent.com/7996307/95014713-f7383500-0648-11eb-8bba-32d4732be4be.png)

div:
![image](https://user-images.githubusercontent.com/7996307/95014723-09b26e80-0649-11eb-8d2f-9ad612c5eb92.png)

## Steps to reproduce

N/A

## Environment

- Home Assistant release with the issue: 0.98 and 0.115.6
- Last working Home Assistant release (if known): < 0.110
- Browser and browser version: Chrome / Safari
- Operating system: Windows / iPad OS

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

**Comment by bramkragten** [comment]
This is not something we support, but I'm willing to take a quick look if you can provide me a simple replication.

**Comment by AmoebeLabs** [comment]
That's nice, I'm happy with all the support/help I can get, as this worked in 0.98.

I will take some time to create a simple example to replicate the problem and report back.

**Comment by AmoebeLabs** [comment]
Sorry, was commenting in wrong issue...

Both configs, 0.98 and 0.115 are using card-mod, although different versions.

I see now that the height of the div in 0.115 is determined by:

![image](https://user-images.githubusercontent.com/7996307/95336460-fefa1280-08b0-11eb-9149-3d9ea952d94a.png)

And the rest of the computed values:
![image](https://user-images.githubusercontent.com/7996307/95336667-38cb1900-08b1-11eb-9041-b29a68e3a9e8.png)


And guess what changes while changing the zoom level in Chrome!!!!

![image](https://user-images.githubusercontent.com/7996307/95336867-7465e300-08b1-11eb-87bb-0bb42464a846.png)

TLDR: Vertical size of div equals:
100% = 20px
150% = 30px
200% = 40px
250% = 50px
300% = 60px
400% = 80px

In hass 0.98, there is no such thing as a line-height defined for that div:

![image](https://user-images.githubusercontent.com/7996307/95337411-184f8e80-08b2-11eb-859d-23adfd79520a.png)

**Comment by AmoebeLabs** [comment]
Yep!

If I remove the line-height using the Chrome inspector, Icons are directly displayed correctly!!

![image](https://user-images.githubusercontent.com/7996307/95338396-2b169300-08b3-11eb-98f7-b2168d4b65ef.png)

So where is this coming from??

**Comment by bramkragten** [comment]
That's a global style. You can overrule it though by setting it yourself.

**Comment by AmoebeLabs** [comment]
Ah!

After some fiddling with syntax the svg rendering works!

```js
return svg`
  <foreignObject width="${this.dimensions.iconSize}em" height="${this.dimensions.iconSize}em" x="${this.coords.xpx}" y="${this.coords.ypx}">
    <div class="div__icon" xmlns="http://www.w3.org/1999/xhtml"
        style="line-height:${this.dimensions.iconSize}em;">
      <ha-icon .icon=${icon} style="${configStyleStr}"></ha-icon>
    </div>
  </foreignObject>
  `;		
```

Me happy! Issue solved & closed.

So this change in global style is something that is inherited somehow by hass using newer libs orso?
